snowfall/decoding/lm_rescore.py

```diff
@@ -187,12 +192,16 @@ def rescore_with_n_best_list(lats: k2.Fsa, G: k2.Fsa,
 
     # Now compute lm_scores
     b_to_a_map = torch.zeros_like(path_to_seq_map)
-    lm_path_lats = _intersect_device(G,
-                                     word_fsas_with_epsilon_loops,
-                                     b_to_a_map=b_to_a_map,
-                                     sorted_match_a=True)
-    lm_path_lats = k2.top_sort(k2.connect(lm_path_lats.to('cpu'))).to(device)
-    lm_scores = lm_path_lats.get_tot_scores(True, True)
+    if G is not None:
+        lm_path_lats = _intersect_device(G,
+                                         word_fsas_with_epsilon_loops,
+                                         b_to_a_map=b_to_a_map,
+                                         sorted_match_a=True)
+        lm_path_lats = k2.top_sort(k2.connect(lm_path_lats.to('cpu'))).to(device)
+        lm_scores = lm_path_lats.get_tot_scores(True, True)
```

Comment:
The 2nd arg to get_tot_scores() here should be false, because ARPA-type language models are constructed in such a way that the backoff prob is included in the direct arc.  I.e. we would be double-counting if we were to sum the probabilities of the non-backoff and backoff arcs.